linters/tslint.js

```diff
@@ -0,0 +1,68 @@
+// Copyright 2019 VMware, Inc.
+// SPDX-License-Identifier: BSD-2-Clause
+
+const cache = require('../cache')
+const report = require('../tslint/tslint_report')
+
+module.exports = class TSLint {
+  get name () {
+    return 'tslint'
+  }
+
+  /**
+   * The name of the generated report file
+   */
+  get reportFile () {
+    return 'tslint.json'
+  }
+
+  get defaultReport () {
+    return { annotations: [], issueCount: { errors: 0, warnings: 0, notices: 0 }, moreInfo: '' }
+  }
+
+  /**
+   * Retains files that can be analyzed by this linter
+   * @param {string[]} files Names of files to analyze
+   * @returns {string[]} Filtered list of file names
+   */
+  filter (files) {
+    return files.filter(name => name.endsWith('.js') || name.endsWith('ts'))
+  }
+
+  /**
+   * Returns the working directory for this analysis
+   * @param {string} repoID Unique repository id
+   * @param {string} prID PR id in repository
+   */
+  workingDirectoryForPR (repoID, prID) {
+    return cache.getBranchPath(repoID, prID, 'tslint')
+  }
+
+  /**
+   * Builds the command line args to pass to the linter process
+   * @param {string[]} files List of files to analyze
+   * @param {string} reportPath Path to the report file relative to working directory
+   */
+  args (files, reportPath) {
+    return ['-c', '../../../tslint/tslint.json', '--format', 'json', '-o', reportPath, ...files]
```

Comment:
What's the 2nd array item here (`'../../../tslint/tslint.json'`), it looks like it would be similar to reportPath?

Reply:
The second argument is the path to the configuration options for TSLint.
Right now we download and process all of the files in the directory
```
cache/<repository_id>/<pull_request_id>
```
and the configuration options for TSLint are in the folder:
```
tslint/tslint.json
```

Reply:
Ah, it's just coincidence that they are very similar? IMHO it would be useful if we didn't use similar filenames/paths for the input/output files. Perhaps we can change `reportFile()` to return _tslint-report.json_ ? We could unify on that filename pattern throughout Precaution? I'm happy for that to be handled separately.

Reply:
That's the first time when we had a coincidence between the output file name and a configuration file.
I will rename the output file to "tslint_output.json".

You know when you brought this up maybe it's a good idea to change the name of the report files for the other linters too.